src/App.tsx

```diff
@@ -1,14 +1,71 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+  useCallback,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
+import { getTodos } from './api';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectedFilter, setSelectedFilter] = useState('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loader, setLoader] = useState(true);
+  const [modalOpened, setModalOpened] = useState(false);
```

Comment:
We can have one state for modal. If there is selected todo, then modal should be open - if no modal should be closed
And when you close modal set selected todo to null